src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -97,115 +92,46 @@ public static class DnnCatalog
-                ReuseValidationSetBottleneckCachedValues = reuseValidationSetBottleneckCachedValues
             };
 
-            if (!File.Exists(options.ModelLocation))
-            {
-                if (options.Arch == Architecture.InceptionV3)
-                {
-                    var baseGitPath = @"https://raw.githubusercontent.com/SciSharp/TensorFlow.NET/master/graph/InceptionV3.meta";
-                    using (WebClient client = new WebClient())
-                    {
-                        client.DownloadFile(new Uri($"{baseGitPath}"), @"InceptionV3.meta");
-                    }
+            return ImageClassification(catalog, options);
+        }
 
-                    baseGitPath = @"https://github.com/SciSharp/TensorFlow.NET/raw/master/data/tfhub_modules.zip";
-                    using (WebClient client = new WebClient())
-                    {
-                        client.DownloadFile(new Uri($"{baseGitPath}"), @"tfhub_modules.zip");
-                        ZipFile.ExtractToDirectory(Path.Combine(Directory.GetCurrentDirectory(), @"tfhub_modules.zip"), @"tfhub_modules");
-                    }
-                }
-                else if(options.Arch == Architecture.ResnetV2101)
-                {
-                    var baseGitPath = @"https://aka.ms/mlnet-resources/image/ResNet101Tensorflow/resnet_v2_101_299.meta";
-                    using (WebClient client = new WebClient())
-                    {
-                        client.DownloadFile(new Uri($"{baseGitPath}"), @"resnet_v2_101_299.meta");
-                    }
-                }
-                else if(options.Arch == Architecture.MobilenetV2)
-                {
-                    var baseGitPath = @"https://tlcresources.blob.core.windows.net/image/MobileNetV2TensorFlow/mobilenet_v2.meta";
-                    using (WebClient client = new WebClient())
-                    {
-                        client.DownloadFile(new Uri($"{baseGitPath}"), @"mobilenet_v2.meta");
-                    }
-                }
-            }
+        /// <summary>
+        /// Performs image classification using transfer learning.
+        /// usage of this API requires additional NuGet dependencies on TensorFlow redist, see linked document for more information.
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!include[io](~/../docs/samples/docs/api-reference/tensorflow-usage.md)]
+        /// ]]>
+        /// </format>
+        /// </summary>
+        /// <param name="catalog"></param>
+        /// <param name="options">An <see cref="Options"/> object specifying advanced options for <see cref="ImageClassificationEstimator"/>.</param>
+        public static ImageClassificationEstimator ImageClassification(
+            this ModelOperationsCatalog catalog, Options options)
+        {
+            options.EarlyStoppingCriteria = options.DisableEarlyStopping ? null : options.EarlyStoppingCriteria ?? new EarlyStopping();
```

Comment:
I still think this acts in a surprising behavior. What if we changed it to the following?

* Remove `options.DisableEarlyStopping`.
* Default `options.EarlyStoppingCriteria = new EarlyStopping();`

Then, by default the EarlyStopping is enabled to the default EarlyStopping. If someone wants to disable EarlyStopping, they set `options.EarlyStoppingCriteria = null;` when calling this API. If they want to change EarlyStopping to a different criteria, they can just set  `options.EarlyStoppingCriteria = new EarlyStopping(metric: EarlyStoppingMetric.Loss)` (or whatever criteria they need).

This feels like a better API design choice because then the caller doesn't have 2 properties to deal with - `DisableEarlyStopping` and `EarlyStoppingCriteria`. There is only 1 to consider, that either they leave to the default, or set it as appropriate. If they want to disable EarlyStopping - they just set it to `null`.